src/components/LmsConfigurations/DegreedIntegrationConfigForm.jsx

```diff
@@ -0,0 +1,292 @@
+          <ValidationFormGroup
+            for="secret"
+            invalid={invalidFields.secret}
+            invalidMessage="Degreed API Client Secret required."
+            helpText="The API Client Secret used to make calls to Degreed on behalf of the customer."
+          >
+            <label htmlFor="secret">API Client Secret</label>
+            <Input
+              type="password"
+              id="secret"
+              name="secret"
+              defaultValue={config ? config.secret : null}
+            />
+          </ValidationFormGroup>
+        </div>
+      </div>
+      <div className="row">
+        <div className="col col-4">
+          <ValidationFormGroup
+            for="degreedUserId"
+            invalid={invalidFields.degreedUserId}
+            invalidMessage="The Degreed User ID is required to access Degreed via Oauth."
+            helpText="The Degreed User ID provided to the content provider by Degreed."
+          >
+            <label htmlFor="degreedUserId">Degreed User ID</label>
+            <Input
+              type="text"
+              id="degreedUserId"
+              name="degreedUserId"
+              defaultValue={config ? config.degreedUserId : null}
+            />
+          </ValidationFormGroup>
+        </div>
+      </div>
+      <div className="row">
+        <div className="col col-4">
+          <ValidationFormGroup
+            for="degreedUserPassword"
+            invalid={invalidFields.degreedUserPassword}
+            invalidMessage="The Degreed User Password is required to access Degreed via Oauth."
+            helpText="The Degreed User Password provided to the content provider by Degreed."
+          >
+            <label htmlFor="degreedUserPassword">Degreed User Password</label>
+            <Input
+              type="password"
+              id="degreedUserPassword"
+              name="degreedUserPassword"
+              defaultValue={config ? config.degreedUserPassword : null}
+            />
+          </ValidationFormGroup>
+        </div>
+      </div>
+
+      <div className="row">
+        <div className="col col-2">
+          <StatefulButton
+            state={submitState}
+            type="submit"
+            id="submitButton"
+            labels={{
+              default: 'Submit',
+              pending: 'Saving...',
+              complete: 'Complete',
+              error: 'Error',
+            }}
+            icons={{
+              default: <Icon className="fa fa-download" />,
+              pending: <Icon className="fa fa-spinner fa-spin" />,
+              complete: <Icon className="fa fa-check" />,
+              error: <Icon className="fa fa-times" />,
+            }}
+            disabledStates={[SUBMIT_STATES.PENDING]}
+            variant="primary"
+            className="ml-3 col"
+          />
+        </div>
+      </div>
+      <div className="row">
+        <div className="col col-3 mt-3">
+          {errorAlert}
```

Comment:
I thought I remembered seeing something about this in the quiz and yeah I finally found it: 
>Error messages should appear at the beginning of the form, and focus should be moved there so that the user can proceed through the form again to correct their mistakes.